### PR TITLE
Default test recording sanitizers

### PR DESCRIPTION
**Pull request by chlowell**
These will apply to all future recordings made through the proxy but don't affect existing recordings. Some of the new target values may not be secret, however we're scrubbing them all anyway to prevent false positives. We can revisit them later if sanitizing them causes any problems.

I marked `NewRecording` deprecated because it doesn't use the test proxy and isn't called anywhere in the repo. We should delete it and the rest of the vestigial `go-vcr` based code because we only support recording via the proxy (#22744).

Also, I reverted #21142 because it pinned our CI pipelines to an old version of the proxy that doesn't support adding sanitizers in bulk, apparently to work around an azdatalake issue. The azdatalake pipelines pass with this change so I guess the underlying problem has been solved? (@scbedd @benbp)

### CODE REVIEW

File: sdk/internal/recording/default_sanitizers.go [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+		return err
+	}
+	req.Body = io.NopCloser(bytes.NewReader(b))
+	req.ContentLength = int64(len(b))
+	req.Header.Set("Content-Type", "application/json")
+	return handleProxyResponse(client.Do(req))
+}
+
+var defaultSanitizers = []sanitizer{
+	newGeneralRegexSanitizer(`("|;)Secret=(?<secret>[^;]+)`, "secret"),
```
Comment (RickWinter) [inline sdk/internal/recording/default_sanitizers.go:100]: Nit: Group the Sanitizers based on 
newGeneral
newBody
newHeader
etc...

File: sdk/internal/recording/recording.go [hunk 2]
```diff
@@ -84,7 +85,12 @@ const (
 	Secret_Base64String VariableType = "secret_base64String"
 )
 
+// defaultSanitizersSet tracks whether default sanitizers have been added (this would be neater with Go 1.19's atomic.Bool)
```
Comment (RickWinter) [inline sdk/internal/recording/recording.go:88]: Lets folow up and move to 1.19. Afterall 1.22.2 and 1.21.9 are the only in support versions on golang.org

**Review by RickWinter — APPROVED** [review]

File: sdk/internal/recording/recording.go (hunk 2 above)
Comment (jhendrixMSFT) [reply sdk/internal/recording/recording.go:88]: Will there be a ripple effect, requiring us to update everything? Might be best to do that separately.

**Pull request merged**